src/enums.h

```diff
@@ -363,4 +363,26 @@ std::array<T, N> vec_to_array(const std::vector<T> &vec)
     return array;
 }
 
+template <typename T>
```

Comment:
Neato! Well honestly, all ya'lls recommends are good. Would it be a better change if I did change it?